boards/shields/nxp_btb44_ov5640/nxp_btb44_ov5640.overlay

```diff
@@ -21,7 +21,7 @@
 
 		port {
 			ov5640_ep_out: endpoint {
-				remote-endpoint = <&mipi_csi2rx_ep_in>;
```

Comment:
Just to make sure I understand here, is the reason we want to use `remote-endpoint-label` over `remote-endpoint` that it avoids the "check-and-egg" dependency problem we have with display and camera drivers? I primarily ask because this deviates from how Linux describes CSI devices in devicetree, right?

Reply:
If the question is for this PR, we need the new video-interfaces bindings for the mipi csi2rx to be able to use the data-lanes property and for the ov5640 to be able to use the bus-type (and data-lanes) property as the driver now supports both CSI2 and DVP modes.

If the question is for the video-interfaces binding (which is merged), replacing remote-endpoint (phandle) by remote-endpoint-label (string) is mandatory as we cannot introduce any binding for endpoint without doing this because it simply does not compile (due to circular dependency).

> I primarily ask because this deviates from how Linux describes CSI devices in devicetree, right?

Apart from the above reason, remote-endpoint-label also helps to bypass the technical obstacle in Zephyr (circular dependency) **to keep Zephyr does NOT deviates from Linux**. In Linux, they don't directly reference the peer devices in DT with direct phandle like we did in Zephyr (with "`source-dev`", "`sensor-dev`", etc). All is done through `remote-endpoint` (some advantages : it's bidirectional, can connect elements at runtime, etc.). And now we can do the same in Zephyr with `remote-endpoint-label`. Someday in the future, if Zephyr supports circular dependency (`remote-endpoint`), we can switch from `remote-endpoint-label` to `remote-endpoint` easily as all the mechanisms will remain the same (it's much easier than the direct phandle reference with `source-dev`, `sensor-dev` we did today).